Formula/bore-cli.rb

```diff
@@ -0,0 +1,23 @@
+class BoreCli < Formula
+  desc "Modern, simple TCP tunnel in Rust that exposes local ports to a remote server"
+  homepage "http://bore.pub"
+  url "https://github.com/ekzhang/bore/archive/v0.2.3.tar.gz"
+  sha256 "c79fa2c7588dec185778bc2bead39265efe6819263ca2a21ca4c24a0221a7949"
+  license "MIT"
+  head "https://github.com/ekzhang/bore.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/bore --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the advise, I'll add some test cases in ~3-4 days when I get back

Reply:
@chenrui333 I am not able to figure out how to add a test for this. Like running `bore local 5000 --to bore.pub` creates a tunnel. There is no config file, no cred failure etc that can be checked :(. Is there any other that opens a tunnel that I can take reference from ?